edx_arch_experiments/config_watcher/signals/receivers.py

```diff
@@ -99,15 +99,24 @@ def _register_waffle_observation(*, model, short_name, fields):
         short_name (str): A short descriptive name for an instance of the model, e.g. "flag"
         fields (list): Names of fields to report on in the Slack message
     """
-    @receiver(signals.post_save, sender=model, dispatch_uid=f"config_watcher_{short_name}_change")
+
+    # Note that weak=False is required here. Django by default only
+    # holds weak references to receiver functions. But these inner
```

Comment:
Out of curiosity, do you know why it stores weak references and what type of clean-up problems this might cause?

Reply:
I'm not sure why it stores weak references, no. I could make some guesses -- perhaps there are architectures where signals are added temporarily, or where an object listens to signals during its lifetime but those listeners should not block garbage collection of the object.

I don't foresee any issue with using strong references here. Normally our `@receiver`-decorated functions are top-level functions, which means the enclosing module would hold a strong reference to the function.

I don't think plugin signals would have had this issue because they require you to point to a top-level function. (Which is also why they were unsuitable for this app, at least if we want to use `sender`. The alternative would be to listen to _all_ model saves in a top-level function and then have a conditional in the receiver's body.)

Reply:
Incidentally, I actually had a branch up for a bit with the top-level function approach before I realized I had overlooked the `weak` param: https://github.com/edx/edx-arch-experiments/compare/main...timmc/top-level#diff-1206f94d4250bda1d46a290ece9b20f6bc8b2f961b6f9adfc91a42002d54099aR93